zerver/context_processors.py

```diff
@@ -190,6 +190,7 @@ def zulip_default_context(request: HttpRequest) -> Dict[str, Any]:
         "only_sso": settings.ONLY_SSO,
         "external_host": settings.EXTERNAL_HOST,
         "external_url_scheme": settings.EXTERNAL_URI_SCHEME,
+        "realm_url": realm_uri,
```

Comment:
This hunk is just used to render templates, and isn't part of the API, so it doesn't belong in this commit and also probably can just be a switch away from `realm_uri`.

Reply:
I guess some hacky API code in `zerver/views/auth.py` reads from this; I ended up tweaking #30022 to use this delta:

```diff
diff --git a/zerver/views/auth.py b/zerver/views/auth.py
index 9c11a695a8..e9cffe568c 100644
--- a/zerver/views/auth.py
+++ b/zerver/views/auth.py
@@ -1120,7 +1120,6 @@ def api_get_server_settings(request: HttpRequest) -> HttpResponse:
     for settings_item in [
         "email_auth_enabled",
         "require_email_format_usernames",
-        "realm_uri",
         "realm_url",
         "realm_name",
         "realm_icon",
@@ -1130,6 +1129,8 @@ def api_get_server_settings(request: HttpRequest) -> HttpResponse:
     ]:
         if context[settings_item] is not None:
             result[settings_item] = context[settings_item]
+    # TODO/compatibility: Backwards-compatibility name for realm_url.
+    result["realm_uri"] = result["realm_url"]
     return json_success(request, data=result)
 
 

```

with the idea being that we won't need to update that file again when we drop `realm_uri` from `context_processors`.